Northstar.CustomServers/mod/scripts/vscripts/mp/_stats.nut

```diff
@@ -12,67 +14,1083 @@ global function PreScoreEventUpdateStats
+		foreach ( entity player in GetPlayerArray() )
+		{
+			Stats_IncrementStat( player, "game_stats", "mode_played", GAMETYPE, 1.0 )
+
+			if ( player.GetTeam() == GetWinningTeam() )
+				Stats_IncrementStat( player, "game_stats", "mode_won", GAMETYPE, 1.0 )
+		}
+	}
+
+	// update player's KD
+	foreach ( entity player in GetPlayerArray() )
+	{
+		// kd stats
+		// index 0 is most recent game
+		// index 9 is least recent game
+		float playerKills = ( player in file.playerKills ) ? file.playerKills[player] : 0.0
+		float playerDeaths = ( player in file.playerDeaths ) ? file.playerDeaths[player] : 0.0
+		float kdratio_match
+		if ( playerDeaths == 0.0 )
+			kdratio_match = playerKills
+		else
+			kdratio_match = playerKills / playerDeaths
+
+		float playerKillsPvp = ( player in file.playerKillsPvp ) ? file.playerKillsPvp[player] : 0.0
+		float playerDeathsPvp = ( player in file.playerDeathsPvp ) ? file.playerDeathsPvp[player] : 0.0
+		float kdratiopvp_match
+		if ( playerDeathsPvp == 0.0 )
+			kdratiopvp_match = playerKillsPvp
+		else
+			kdratiopvp_match = playerKillsPvp / playerDeathsPvp
+
+		float totalDeaths = player.GetPersistentVarAsInt( "deathStats.total" ).tofloat()
+		float totalKills = player.GetPersistentVarAsInt( "killStats.total" ).tofloat()
+		float totalDeathsPvp = player.GetPersistentVarAsInt( "deathStats.totalPVP" ).tofloat()
+		float totalKillsPvp = player.GetPersistentVarAsInt( "killStats.totalPVP" ).tofloat()
+		float kdratio_lifetime
+		if ( totalDeaths == 0.0 )
+			kdratio_lifetime = totalKills
+		else
+			kdratio_lifetime = totalKills / totalDeaths
+		float kdratio_lifetimepvp
+		if ( totalDeathsPvp == 0.0 )
+			kdratio_lifetimepvp = totalKillsPvp
+		else
+			kdratio_lifetimepvp = totalKillsPvp / totalDeathsPvp
+		
+		// shift stats by 1 to make room for new game data
+		for ( int i = NUM_GAMES_TRACK_KDRATIO - 2; i >= 0; --i )
+		{
+			player.SetPersistentVar( "kdratio_match[" + ( i + 1 ) + "]", player.GetPersistentVar( "kdratio_match[" + i + "]" ) )
+			player.SetPersistentVar( "kdratiopvp_match[" + ( i + 1 ) + "]", player.GetPersistentVar( "kdratiopvp_match[" + i + "]" ) )
+		}
+		// add new game data
+		player.SetPersistentVar( "kdratio_match[0]", kdratio_match )
+		player.SetPersistentVar( "kdratiopvp_match[0]", kdratiopvp_match )
+		player.SetPersistentVar( "kdratio_lifetime", kdratio_lifetime )
+		player.SetPersistentVar( "kdratio_lifetime_pvp", kdratio_lifetimepvp )
+	}
+
+	// award mvp and top 3 in each team
+	if ( !IsFFAGame() )
+	{
+		string gamemode = GameRules_GetGameMode()
+		int functionref( entity, entity ) compareFunc = GameMode_GetScoreCompareFunc( gamemode )
+
+		for( int team = 0; team < MAX_TEAMS; team++ )
+		{
+			array<entity> players = GetPlayerArrayOfTeam( team )
+			if ( compareFunc == null )
+			{
+				printt( "gamemode doesn't have a compare func to get the top 3" )
+				return
+			}
+			players.sort( compareFunc )
+			int max = min( players.len(), 3 ).tointeger()
```

Comment:
```suggestion
			int max = int( min( players.len(), 3 ) )
```

Reply:
The compiler might be able to return as int here itself bc the input and output should be just numeric.
My question why this even compiles would be rather why `max` is a valid identifier when a native function like that already exists

Reply:
> My question why this even compiles would be rather why `max` is a valid identifier when a native function like that already exists

that's a good point lol, I'll change that variable name (even tho it does compile)